ftplugin/javascript/cfi.vim

```diff
@@ -0,0 +1,181 @@
+scriptencoding utf-8
+
+" ============================================================================
+" Bootstrap
+" ============================================================================
+
+if get(g:, 'cfi_disable') || get(g:, 'loaded_cfi_ftplugin_javascript')
+    finish
+endif
+let g:loaded_cfi_ftplugin_javascript = 1
+
+" Saving 'cpoptions' {{{
+let s:save_cpo = &cpo
+set cpo&vim
+" }}}
+
+" ============================================================================
+" Options
+" ============================================================================
+
+if !exists('g:cfi_javascript_show_params')
+    let g:cfi_javascript_show_params = 0
+endif
+
+" ============================================================================
+" Regexes
+" magic mode!
+" ============================================================================
+
+" anystring
+let s:FUNCTION_NAME       = '\(\S\+\)'
+
+" matches es6 classes and generator functions
+" @TODO es6 member function shorthand `func() {}`
+let s:FUNCTION_TYPE       = '\(get\|set\|static\|function\*\=\)'
```

Comment:
added better match
https://github.com/davidosomething/current-func-info.vim/commit/4195d71e76eb0dfaed208f8fae622d8059703e48#diff-28afe41f9b859240481b7b38ccc8aad5R30

and also clean up if using strange spacing:
https://github.com/davidosomething/current-func-info.vim/commit/4195d71e76eb0dfaed208f8fae622d8059703e48#diff-28afe41f9b859240481b7b38ccc8aad5R129